### PR TITLE
fix and improve messages grouping

### PR DESCRIPTION
**Pull request by Dinek007**

### CODE REVIEW

**Comment by holmesworcester** [comment]
@Dinek007 this should have tests to catch basic issues as well as the bug we're fixing now (messages from different users get grouped).

File: src/renderer/components/widgets/channels/ChannelMessage.js [hunk 1]
```diff
@@ -276,70 +276,91 @@ export const ChannelMessage = ({
     }
   }, [imageUrl])
   const [actionsOpen, setActionsOpen] = useState(false)
+  const arrayOfGroupedMessages = [[]]
+  for (const el of parsedMessage) {
```
Comment (kowalski) [inline src/renderer/components/widgets/channels/ChannelMessage.js:280]: we shouldn't do this logic in render;

imo move this to the effect that is calling setParsedMessage

**Review by kowalski — CHANGES_REQUESTED** [review]

**Review by kowalski — APPROVED** [review]

**Review by siepra — APPROVED** [review]

File: src/renderer/components/widgets/channels/ChannelMessage.tsx [hunk 2]
```diff
@@ -255,49 +254,68 @@ export const ChannelMessage: React.FC<ChannelMessageProps> = ({
...
       <Grid className={classes.messageInput} item>
-        <Typography variant='body2' className={classes.message}>
-          {parsedMessage}
-        </Typography>
-        {/* {status === 'failed' && (
-          <ChannelMessageActions onResend={() => onResend(message)} />
-        )} */}
+        {
+          allMessages.map((item) => {
+            if (item !== ' ') {
```
Comment (holmesworcester) [inline src/renderer/components/widgets/channels/ChannelMessage.tsx:306]: In what case is item == ' ' ?

File: src/renderer/store/selectors/channel.ts [hunk 3]
```diff
@@ -136,31 +133,22 @@ const concatMessages = (mainMsg, messagesToConcat) => {
...
-
+    } else if (lastCteatedAt && ((lastCteatedAt - msg.createdAt) < timeOfStackMessages)) {
       result[result.length - 1].push(msg)
     } else {
+      result.push([])
       result[result.length - 1].push(msg)
     }
-    last = msg
+    lastPubKey = msg.pubKey
+    lastCteatedAt = msg.createdAt
```
Comment (holmesworcester) [inline src/renderer/store/selectors/channel.ts:151]: Does this approach allow us to style grouped messages differently than, say, a single message with a bunch of line breaks in it? 

Will it let us put controls on individual messages, say, if you wanted to delete a single message within a series of grouped messages?

**Review by holmesworcester — APPROVED** [review]

**Pull request merged**